src/PKSim.Core/Services/SimulationParametersToBuildingBlockUpdater.cs

```diff
@@ -59,17 +59,16 @@ public ICommand UpdateParametersFromSimulationInBuildingBlock(Simulation simulat
          updateCommands.BuildingBlockName = templateBuildingBlock.Name;
          updateCommands.CommandType = PKSimConstants.Command.CommandTypeUpdate;
          updateCommands.Description = PKSimConstants.Command.UpdateTemplateBuildingBlockCommandDescription(buildingBlockType, templateBuildingBlock.Name, simulation.Name);
-         _executionContext.UpdateBuildinBlockProperties(updateCommands, templateBuildingBlock);
+         _executionContext.UpdateBuildinBlockPropertiesInCommand(updateCommands, templateBuildingBlock);
 
          _executionContext.PublishEvent(new BuildingBlockUpdatedEvent(templateBuildingBlock));
          return updateCommands;
       }
 
-
-
-      private IPKSimCommand updateParameterValues(PathCache<IParameter> sourceParameters, PathCache<IParameter> targetParameters)
+      private IOSPSuiteCommand updateParameterValues(PathCache<IParameter> simulationParameters, PathCache<IParameter> templateParameters)
       {
-         return _parameterSetUpdater.UpdateValues(sourceParameters, targetParameters).DowncastTo<IPKSimCommand>();
+         //we do not want to update parameter origin id here since we are updateing building block parameters
+         return _parameterSetUpdater.UpdateValues(simulationParameters, templateParameters, updateParameterOriginId: false);
```

Comment:
This was  bug in current PK-Sim implementation. Without this flag (new flag) the OriginParameterId of a template building block was set to the ParameterId of teh parameter in the simulation after Commit. This could have lead potentially to wrong references being updated